src/language/parser.js

```diff
@@ -152,23 +152,26 @@ export function parse(
  * in isolation of complete GraphQL documents.
  */
 export function parseValue(
-  source: Source | string,
+  source: string | Source,
   options?: ParseOptions
 ): Value {
-  const sourceObj = source instanceof Source ? source : new Source(source);
-  const parser = makeParser(sourceObj, options || {});
-  return parseValueLiteral(parser, false);
+  const sourceObj = typeof source === 'string' ? new Source(source) : source;
+  const lexer = createLexer(sourceObj, options || {});
+  expect(lexer, TokenKind.SOF);
+  const value = parseValueLiteral(lexer, false);
+  expect(lexer, TokenKind.EOF);
```

Comment:
This was an existing bug! Previously `parseValue('123garbage')` would return the value `123`, now it rightfully creates a parse error.